### PR TITLE
Minor fixes for SVMLight and LibSvm record readers and writers

### PR DESCRIPTION
**Pull request by turambar**
## What changes were proposed in this pull request?

(1) added support for records with no labels, either fully unlabeled data or multilabel with no positive labels
(2) for features and sparse multilabels, now pre-allocate List<Writables> with Collections.numCopies() and call set(index, value) to update features, rather than using ad hoc while loops.

Added new unit tests or updated existing unit tests to cover various "no label" scenarios. All unit tests (including legacy tests) pass.

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

Please review
https://github.com/deeplearning4j/deeplearning4j/blob/master/CONTRIBUTING.md before opening a pull request.

### CODE REVIEW

**Comment by turambar** [comment]
TL;DR previously, support for records with no labels (either fully unlabeled data or multilabel records with no positive labels) was incomplete. Now we cover most common scenarios, and we're now consistent with scikit-learn's functionality and behavior.
